draft-private-access-tokens.md

```diff
@@ -385,65 +402,120 @@ listing both the ANON_CLIENT_ID as "Anonymous-Client-ID", and the ANON_ORIGIN_ID
 :path = /access-token-request
 accept = message/ohttp-req
 cache-control = no-cache, no-store
-content-type = message/ohttp-req
-content-length = ...
+content-type = message/access-token-request
+content-length = 512
+Token-Origin = ENCRYPTED_ORIGIN_NAME
 Anonymous-Origin-ID = ANON_ORIGIN_ID_PRIME
 Anonymous-Client-ID = ANON_CLIENT_ID
 
-<Bytes containing the encapsulated HTTP message for the Issuer>
+<Bytes containing the AccessTokenRequest>
 ~~~
 
 Note: although these headers are per-request, they do not reveal information about
-the originating client. See {{client-id}} and {{origin-id}} for more details.
-
-Upon receipt of the Client's encapsulated request, the Issuer checks for the "Anonymous-Client-ID"
-and "Anonymous-Origin-ID" headers. If either is absent, the Issuer aborts and returns a 400 error
-to the Mediator. If present, the Issuer decapsulates the request. If this fails, the Issuer aborts
-and returns a 400 error to the Mediator. If decapsulation succeeds, the Issuer checks for the
-"Target-Origin" header. If absent, the Issuer aborts and returns a 400 error to the Mediator.
-If present, the Issuer proceeds extracts ANON_CLIENT_ID from the "Anonymous-Client-ID" header,
-ANON_ORIGIN_ID from the "Anonymous-Origin-ID" header, and ORIGIN_NAME from the "Target-Origin"
-header, and then proceeds as follows.
-
-First, check to see if there are any prior token requests for the given (ANON_CLIENT_ID, ORIGIN_NAME) pair.
-If so, and if the corresponding ANON_ORIGIN_ID does not match that of the current request, the
-Issuer aborts and returns a 400 error to the Mediator.
-
-If this is not the case, determine if the token request can be satisfied for the given
-(ANON_CLIENT_ID, ORIGIN_NAME) pair, according to the access token policy. If the policy does not admit
-issuance, the Issuer aborts and returns a 400 error to the Mediator.
-
-If the Issuer local state and policy admit a token, the Issuer completes the issuance flow by
+the originating Client. See {{client-id}} and {{origin-id}} for more details.
+
+Upon receipt of the forwarded request, the Issuer validates the following
+conditions:
+
+- The "Anonymous-Client-ID" header is present
+- The "Anonymous-Origin-ID" header is present
+- The "Token-Origin" header is present, and can be decrypted using the Issuer's private key
+(the private key associated with ISSUER_NAME_KEY).
+- The decrypted ORIGIN_NAME matches an Origin the Issuer serves
+
+If any of these conditions is not met, the Issuer MUST return an HTTP 400 error to the Mediator,
+which will forward the error to the client.
+
+If the conditions are met, the Issuer then applies its policy for the Client request, by checking
+for state associated with the ANON_CLIENT_ID. If there is previous state associated with the
+ANON_CLIENT_ID, the Issuer first validates two conditions:
+
+- The decrypted ORIGIN_NAME for this request matches the ORIGIN_NAME stored from previous requests
```

Comment:
Should this instead just check for the unique tuple of {ANON_CLIENT_ID, ORIGIN_NAME}? A collision is highly unlikely, but I think this makes the case less confusing.